module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AuditEventsTest.php

```diff
@@ -0,0 +1,296 @@
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+final class AuditEventsTest extends \VuFindTest\Integration\MinkTestCase
+{
+    use \VuFindTest\Feature\LiveDatabaseTrait;
+    use \VuFindTest\Feature\UserCreationTrait;
+    use \VuFindTest\Feature\DemoDriverTestTrait;
+
+    /**
+     * Standard setup method.
+     *
+     * @return void
+     */
+    public static function setUpBeforeClass(): void
+    {
+        static::failIfDataExists();
+    }
+
+    /**
+     * Test audit events disabled.
+     *
+     * @return void
+     */
+    public function testEventsDisabled(): void
+    {
+        // Setup config
+        $this->changeConfigs(
+            [
+                'Demo' => [
+                    'Users' => ['username1' => 'catpass'],
+                ],
+                'config' => [
+                    'Catalog' => ['driver' => 'Demo'],
+                    'Authentication' => [
+                        'method' => 'ILS',
+                    ],
+                    'Logging' => [
+                        'log_audit_events' => '',
+                    ],
+                ],
+            ]
+        );
+
+        // Purge events:
+        $eventService = $this->getDbService(AuditEventService::class);
+        $eventService->purgeEvents();
+
+        // Log in:
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl('/MyResearch/Profile'));
+        $page = $session->getPage();
+        $this->findCssAndSetValue($page, '#login_ILS_username', 'username1');
+        $this->findCssAndSetValue($page, '#login_ILS_password', 'catpass');
+        $this->clickCss($page, 'input.btn.btn-primary');
+
+        // Log out:
+        $this->clickCss($page, '.logoutOptions a.logout');
+
+        // Check events:
+        $events = $eventService->getEvents();
+        $this->assertEmpty($events);
+    }
+
+    /**
+     * Test login events.
+     *
+     * @return void
+     */
+    public function testLoginEvents(): void
+    {
+        // Setup config
+        $this->changeConfigs(
+            [
+                'Demo' => [
+                    'Users' => ['username2' => 'catpass'],
+                ],
+                'config' => [
+                    'Catalog' => ['driver' => 'Demo'],
+                    'Authentication' => [
+                        'method' => 'ILS',
+                        'account_deletion' => true,
+                    ],
+                    'Logging' => [
+                        'log_audit_events' => 'ils,user',
+                    ],
+                ],
+            ]
+        );
+
+        // Purge events:
+        $eventService = $this->getDbService(AuditEventServiceInterface::class);
```

Comment:
Why do you retrieve using the interface here but the service elsewhere? I would think consistency would be better.